app/code/community/Drip/Connect/Model/Observer/Customer/GuestSubscriberCreated.php

```diff
@@ -9,15 +9,26 @@ class Drip_Connect_Model_Observer_Customer_GuestSubscriberCreated extends Drip_C
      */
     protected function executeWhenEnabled($observer)
     {
-        if (! Mage::registry(self::REGISTRY_KEY_NEW_GUEST_SUBSCRIBER)) {
-            return;
-        }
-
         $email = Mage::app()->getRequest()->getParam('email');
+
         $subscriber = Mage::getModel('newsletter/subscriber')->loadByEmail($email);
 
         $config = Drip_Connect_Model_Configuration::forCurrentScope();
 
+        if (! Mage::registry(self::REGISTRY_KEY_NEW_GUEST_SUBSCRIBER)) {
+            $customer = Mage::helper('drip_connect')->getCustomerByEmail($email);
+            if ((bool) $customer->getId()) {
+              Mage::helper('drip_connect/customer')->proceedAccount(
+                $customer,
+                $config,
+                true,
+                Drip_Connect_Model_ApiCalls_Helper_RecordAnEvent::EVENT_CUSTOMER_UPDATED,
+                true
+              );
+            }
+            return;
```

Comment:
Nit: When the bulk of the logic is inside an if statement, I prefer to use else here rather than a guard clause. I think it's easier to see what's going on.